crates/provider-keyvalue-redis/src/lib.rs

```diff
@@ -381,6 +456,179 @@ impl Provider for KvRedisProvider {
+                anyhow::anyhow!("Unexpected response format from Redis CONFIG GET")
+            })?;
+
+            if !current_config.contains('K')
+                || !current_config.contains('$')
+                || !current_config.contains('g')
+            {
+                error!(
+                    current_config = %current_config,
+                    "Redis keyspace-notifications not properly configured"
+                );
+                return Err(anyhow::anyhow!(
+                    "Redis keyspace-notifications not properly configured! \
+                        Expected 'K$g' in settings, but got '{}'. \
+                        Please run: CONFIG SET notify-keyspace-events K$g",
+                    current_config
+                ));
+            }
+
+            let wrpc = Arc::new(wrpc);
+            let wrpc_for_task = wrpc.clone();
+
+            let config_watch_entries = parse_watch_config(&config, &target_id);
+
+            // Update watched keys
+            let mut watched_keys = self.watched_keys.write().await;
+            for (key, key_info_set) in config_watch_entries {
+                watched_keys
+                    .entry(key)
+                    .or_insert_with(HashSet::new)
+                    .extend(key_info_set);
+            }
+
+            let client_clone = client.clone();
+            let self_clone = self.clone();
+            let mut conn_clone = conn.clone();
+            let task = tokio::spawn(async move {
+                let mut pubsub = match client_clone.get_async_pubsub().await {
+                    Ok(pubsub) => pubsub,
+                    Err(e) => {
+                        error!(err = %e, "Failed to get pubsub connection");
+                        return;
+                    }
+                };
+                let watched_keys = self_clone.watched_keys.read().await;
+                for key in watched_keys.keys() {
+                    let channel = format!("__keyspace@0__:{}", key);
+                    let _ = pubsub
+                        .psubscribe(&channel)
+                        .await
+                        .context("Failed to subscribe to SET/DEL events for key");
+                }
+                let stream = pubsub.on_message();
+                tokio::pin!(stream);
+                while let Some(msg) = stream.next().await {
+                    let channel: String = msg.get_channel_name().to_string();
+                    let event: String = match msg.get_payload() {
+                        Ok(event) => event,
+                        Err(e) => {
+                            error!(err = %e, "Failed to get payload");
+                            continue;
+                        }
+                    };
+                    // The Channel is in the format __keyspace@0__:key
+                    // While the payload is the event (ie set | del)
+                    let mkey = channel.split(':').last().unwrap();
```

Comment:
This should log an error or warning if the split doesn't work rather than unwrapping to avoid a panic